### DISCUSSION TITLE
Use case: Power off if grid carbon intensity is high? Pre-heat/cool if carbon forecast is increasing?

### DISCUSSION THREAD

**Discussion by ando2040** [revk/xESP32-Faikout#167] [Q&A]
I'd like to query a web source, preferably every 30 min, and take action eg power off if carbon intensity on the grid is high or power prices are increasing.

I had a couple of thoughts on how to achieve this, none of which has worked out so far. I think I'm heading the Faikin route (Daikin have desupported IFTTT which was one idea) but I haven't installed one yet. Just wondered if anyone in this community has already worked through pulling data from a web source (in this case probably [carbonintensity.org.uk](https://carbonintensity.org.uk/) , maybe Octopus Agile API) and happy to share their setup please?

I'm not a high skill coder but I can probably link components together. I'd mainly like to be reassured it can be done and if possible have an example to work from.

**Comment by revk** [comment]
Probably not complex. Easy to tell the Faikin to turn off. I expect a script on a raspberry pi or something. I wonder if Home Assistant has a plug in for this as you might want lots of things to turn off or turn down...

**Reply by ando2040** [reply]
Good thought - indeed there is!
https://community.home-assistant.io/t/carbon-intensity-uk/473199

I wanted to keep it simple but this does look impressive so I will investigate.

**Reply by ando2040** [reply]
That integration was not as impressive as at first sight. However, I went with this one which works well, using high price as a proxy for high carbon (which holds fairly true, and if I switch to this tariff, I will save money as well every time it reduces carbon load):
https://community.home-assistant.io/t/octopus-agile-display-tariff-in-graphs-tables-best-import-export-periods-all-done-using-node-red-and-jsonata/518762

I'll confess that I haven't actually installed the Faikin module yet; I found a Home Assistant integration that pretends to be an app user for the time being. I do have a problem with the Perfera (FTXM-R) temperature sensors in heating mode having a huge offset (11C this morning! More commonly 5-6C) that will probably be my reason to install the Faikin module (with external Mi sensor) which I bought for testing.

It's off topic but I'd be interested to hear if others have had problems with integrated temperature sensor offset from reality. Daikin insist it should be reliable and they have no idea what I'm talking about...

**Comment by 44616E** [comment]
I just use Nodered and mqtt, then its trivial to access internet data sources and control the Faikins.
